frontend/src/pages/projects/projectSharing/ProjectSharingTableRow.tsx

```diff
@@ -0,0 +1,172 @@
+  );
+  const [isOpen, setIsOpen] = React.useState(false);
+  const [isOpenName, setIsOpenName] = React.useState(false);
+  const [isLoading, setIsLoading] = React.useState(false);
+
+  return (
+    <Tbody>
+      <Tr>
+        <Td dataLabel="Username">
+          {isEditing ? (
+            <Select
+              variant={SelectVariant.typeahead}
+              typeAheadAriaLabel="Name selection"
+              selections={roleBindingName}
+              onToggle={(isOpened) => {
+                setIsOpenName(isOpened);
+              }}
+              onSelect={(e, selection) => {
+                if (typeof selection === 'string') {
+                  setRoleBindingName(selection);
+                  setIsOpenName(false);
+                }
+              }}
+              onClear={() => setRoleBindingName('')}
+              isOpen={isOpenName}
+              isCreatable={true}
+              aria-labelledby="name-selection"
+              placeholderText={roleBindingName}
+            >
+              {typeAhead.map((option, index) => (
+                <SelectOption key={index} value={option} />
+              ))}
+            </Select>
+          ) : (
+            <Text>{roleBindingName}</Text>
+          )}
+        </Td>
+        <Td dataLabel="Permission">
+          {isEditing ? (
+            <Select
+              removeFindDomNode
+              selections={roleBindingRoleRef}
+              isOpen={isOpen}
+              onSelect={(e, selection) => {
+                if (typeof selection === 'string') {
+                  setRoleBindingRoleRef(
+                    castProjectSharingRoleType(selection) || ProjectSharingRoleType.EDIT,
+                  );
+                  setIsOpen(false);
+                }
+              }}
+              onToggle={setIsOpen}
+              placeholderText={roleBindingRoleRef}
+              direction="down"
+            >
+              <SelectOption
+                key={ProjectSharingRoleType.EDIT}
+                value={ProjectSharingRoleType.EDIT}
+                description={'Edit allows users to view and make changes to the project'}
+              >
+                {ProjectSharingRoleType.EDIT}
+              </SelectOption>
+              <SelectOption
+                key={ProjectSharingRoleType.ADMIN}
+                value={ProjectSharingRoleType.ADMIN}
+                description={'Admin allows users to also add and remove new users to the project'}
+              >
+                {ProjectSharingRoleType.ADMIN}
+              </SelectOption>
+            </Select>
+          ) : (
+            <Text> {roleBindingRoleRef}</Text>
+          )}
+        </Td>
+        <Td dataLabel="Date added">
+          {!isEditing && obj && (
+            <Text>
+              {obj.metadata?.creationTimestamp ? formatDate(obj.metadata?.creationTimestamp) : ''}
+            </Text>
+          )}
+        </Td>
+        {!isEditing && obj && (
```

Comment:
```suggestion
        {!isEditing && (
```
Avoid conditions that cannot be true -- we have TS. Your component type said `obj` was required. Also, consider not even caring about `obj` and let the caller deal with it. It's a contextual component, the caller knows what they passed.